### PR TITLE
fix: ensure all Person instances have valid ids

### LINKED ISSUE #812: regression: mailto:None as id of dataset creators
`mailto:None` appears as a node ID of dataset creators. 

This was fixed but the regression came about during the refactoring of `Creator` and `Person` classes.

### PR DESCRIPTION
**Pull request by rokroskar**
closes #812

### CODE REVIEW

**Comment by rokroskar** [comment]
yep, fixing the lowercase issue :)

your tests should be there

File: renku/core/models/datasets.py [hunk 1]
```diff
@@ -44,6 +45,32 @@
...
+    if isinstance(value, dict):  # compatibility with previous versions
+        return [Person.from_jsonld(value)]
+
+    if isinstance(value, list):
+        return [Person.from_jsonld(v) for v in value]
+
+
+@attr.s
+class PersonMixin:
+    """Mixin for handling creators container."""
```
Comment (Panaetius) [inline renku/core/models/datasets.py]: Maybe this should still be named `CreatorsMixin` as it is for setting the creator on an object. Especially given the docstring there.

**Comment by Panaetius** [comment]
Thanks a lot for this change! it makes a lot of sense to get rid of `Creator`.

File: renku/core/models/datasets.py (hunk 1 above)
Comment (rokroskar) [reply renku/core/models/datasets.py]: good point - I changed it back

**Pull request merged**